tests/framework/extensions/provisioning/ssh.go

```diff
@@ -8,19 +8,25 @@ import (
 	"errors"
 	"strconv"
 	"strings"
+	"time"
 
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	nodev1 "github.com/rancher/rancher/tests/framework/extensions/nodes"
 	"github.com/rancher/rancher/tests/framework/pkg/nodes"
 	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/util/wait"
 )
 
 const (
 	cpuUsageVar = 100 // 100 is just a placeholder until we can determine an actual number. Even with cpu usage spiking it should not go past 100% cpu usage and previous issues concerning this were hitting around 130% and above
 	checkCPU    = "CheckCPU"
+	nodeReboot  = "NodeReboot"
+	activeState = "active"
 )
 
 // This func tests the ssh tests specified in the provisioninginput config clusterSSHTests field.
 // For example CheckCPU checks the cpu usage of the cluster agent. If the usage is too high the func will return a warning.
-func CallSSHTestByName(testName string, node *nodes.Node) error {
+func CallSSHTestByName(testName string, node *nodes.Node, client *rancher.Client, clusterID string) error {
```

Comment:
These should really be string enums
```suggestion
type SshTestCase string
const (
	cpuUsageVar = 100 // 100 is just a placeholder until we can determine an actual number. Even with cpu usage spiking it should not go past 100% cpu usage and previous issues concerning this were hitting around 130% and above
	activeState = "active"
	checkCPU    SshTestCase = "CheckCPU"
	nodeReboot  						= "NodeReboot"
)

// This func tests the ssh tests specified in the provisioninginput config clusterSSHTests field.
// For example CheckCPU checks the cpu usage of the cluster agent. If the usage is too high the func will return a warning.
func CallSSHTestByName(testCase SshTestCase, node *nodes.Node, client *rancher.Client, clusterID string) error {
```